torch/_inductor/lowering.py

```diff
@@ -4276,10 +4276,11 @@ def adaptive_max_pool2d(x, output_size):
         return empty(o_size, dtype=x.get_dtype(), device=x.get_device()), empty(
             o_size, dtype=torch.int64, device=x.get_device()
         )
+
     if h_in % h_out == 0 and w_in % w_out == 0:
         kernel_size = [h_in // h_out, w_in // w_out]
         if should_fallback_max_pool2d_with_indices(kernel_size, dilation=[1, 1]):
-            return max_pool2d_with_indices(x, kernel_size)  # type: ignore[name-defined]   # noqa: F821
+            return aten.max_pool2d_with_indices(x, kernel_size)  # type: ignore[name-defined]   # noqa: F821
```

Comment:
@amjames, should we just call this unconditionally?

Reply:
FWIW I think using a properly defined fallback handler would work here too